inner-scripts/kubestellar

```diff
@@ -38,54 +38,98 @@ remove=0
 cleanup=0
 espw_name="espw"
 log_folder=$(pwd)/kubestellar-logs
-local_kcp=true
 imws="root:imw1"
 wmws="root:wmw1"
 
 function echoerr() {
    echo "ERROR: $1" >&2
 }
 
+function create_space() { # usage: space_name
+    space_name=$1
+    kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} create -f - <<EOF
+apiVersion: space.kubestellar.io/v1alpha1
+kind: Space
+metadata:
+  name: $space_name
+  namespace: "${PROVIDER_NAMESPACE}"
+spec:
+  SpaceProviderDescName: "${PROVIDER_NAME}"
+  Type: "managed"
+EOF
+    echo "Waiting for space $space_name to get into the Ready phase."
+    until [ "$(kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get space $space_name -n ${PROVIDER_NAMESPACE} -o yaml | grep Ready)" != "" ]; do
+        sleep 1
+    done
+}
+
+# TODO: caching support - the kubeconfig file could already be in the kubeconfig directory.
+function get_kubeconfig_file() { # usage: space_name provider_namespace
+    space_name="$1"
+    provider_namespace="$2"
+    if [ "$IN_CLUSTER" == "true" ] ; then
+        secret_name=`kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get space $space_name -n $provider_namespace -o jsonpath='{$.status.inClusterSecretRef.name}'`
+        secret_namespace=`kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get space $space_name -n $provider_namespace -o jsonpath='{$.status.inClusterSecretRef.namespace}'`
+    else
+        secret_name=`kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get space $space_name -n $provider_namespace -o jsonpath='{$.status.externalSecretRef.name}'`
+        secret_namespace=`kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get space $space_name -n $provider_namespace -o jsonpath='{$.status.externalSecretRef.namespace}'`
+    fi
+    kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get secret $secret_name -n $secret_namespace -o jsonpath='{$.data.kubeconfig}' | base64 -d | base64 -d > ${KUBECONFIG_DIR}/${provider_namespace}-${space_name}
+}
+
+function create_space_if_needed_and_get_config() { # usage: space_name
+    space_name="$1"
+    if kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} get space $space_name -n ${PROVIDER_NAMESPACE} 2>&1 | grep "not found" ; then
+        create_space $space_name
+    fi
+    get_kubeconfig_file $space_name ${PROVIDER_NAMESPACE} 
+} 
+ 
+function run_kubectl_in_space() { # usage: space_name kubectl_cmd
+    space_name="$1"
+    cmd="$2"
+    kubeconfig_path="${KUBECONFIG_DIR}/${PROVIDER_NAMESPACE}-$space_name"
+    if [ -f $kubeconfig_path ] ; then 
```

Comment:
Don't you want the opposite of this condition here? Fetch the file if it is NOT already present?

Reply:
Why was this marked as resolved?